src/components/app/details/appDetails/utils.tsx

```diff
@@ -384,127 +386,233 @@ export function addQueryParamToGrafanaURL(url: string, appId: string | number, e
+                            tableData.currentPhase = phase
+                            if(item.resourceStatus === 'failed'){
+    
+                            }
+                            tableData.currentTableData.push({icon: 'success',phase: phase, message: `${phase}: Create and update resources based on manifest` })
+                            return
+                        }
+                    }
+                  })
+              }
               if (
-                  element['status'] === 'KUBECTL_APPLY_STARTED' &&
+                  element['status'] === TIMELINE_STATUS.KUBECTL_APPLY_STARTED &&
                   deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.time === '' &&
-                  deploymentData.deploymentStatus !== 'Succeeded'
+                  deploymentData.deploymentStatus !== DEPLOYMENT_STATUS.SUCCEEDED
               ) {
-                  if (deploymentData.deploymentStatus === 'failed') {
+                deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.resourceDetails = element.resourceDetails.filter((item) => item.resourcePhase ===  tableData.currentPhase)//// check this condition
+                  if (deploymentData.deploymentStatus === DEPLOYMENT_STATUS.FAILED) {
                       deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.icon = 'unknown'
                       deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.displaySubText = ': Unknown'
                       deploymentData.deploymentStatusBreakdown.APP_HEALTH.icon = 'unknown'
                       deploymentData.deploymentStatusBreakdown.APP_HEALTH.displaySubText = ': Unknown'
-                  } else if (deploymentData.deploymentStatus === 'succeeded') {
+                  } else if (deploymentData.deploymentStatus === DEPLOYMENT_STATUS.SUCCEEDED) {
                     deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.icon = 'success'
                     deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.displaySubText = ''
+                } else if (deploymentData.deploymentStatus === DEPLOYMENT_STATUS.TIMED_OUT) {
+                    deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.icon = 'unknown'
+                    deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.isCollapsed = false
+                    deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.displaySubText = 'Unknown'
+                    deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.timelineStatus = deploymentData.deploymentError
+                    deploymentData.deploymentStatusBreakdown.KUBECTL_APPLY.kubeList = tableData.currentTableData.map((item) => {
+                        return {
+                            icon: item.phase === tableData.currentPhase ? 'failed' : 'success',
+                            message: item.message
+                        }
+                      })
+                }  else if (deploymentData.deploymentStatus === DEPLOYMENT_STATUS.UNABLE_TO_FETCH) {
```

Comment:
can we merge `UNABLE_TO_FETCH` & `TIMED_OUT`?

Reply:
done